src/btree/bt_page.c

```diff
@@ -610,9 +609,11 @@ __inmem_row_leaf(WT_SESSION_IMPL *session, WT_PAGE *page)
             __wt_row_leaf_value_cell(session, page, rip, NULL, &unpack);
             if (unpack.tw.prepare) {
                 /* Take the value from the original page cell. */
-                WT_RET(__wt_page_cell_data_ref(session, page, &unpack, &buf));
+                if (value == NULL)
+                    WT_RET(__wt_scr_alloc(session, 0, &value));
+                WT_RET(__wt_page_cell_data_ref(session, page, &unpack, value));
```

Comment:
Converting the following two from WT_RET to WT_ERR can handle freeing the memory in case of failure, otherwise, they will return.

Reply:
Thanks! Fixed in 79686c59d.